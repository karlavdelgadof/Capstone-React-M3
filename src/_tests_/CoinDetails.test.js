import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import CoinDetails from '../pages/CoinDetails';
import store from '../redux/configureStore';

describe('Coins test', () => {
  test('snapshot for coins', () => {
    const history = createBrowserHistory();
    const coin = {
      id: '90',
      symbol: 'BTC',
      name: 'Bitcoin',
      nameid: 'bitcoin',
      rank: 1,
      price_usd: '20142.07',
      percent_change_24h: '1.44',
      percent_change_1h: '-0.36',
      percent_change_7d: '-5.64',
      price_btc: '1.00',
      market_cap_usd: '385002452610.93',
      volume24: '27391301861.27469',
      volume24a: '28376681317.71335',
      csupply: '19114346.00',
      tsupply: '19114346',
      msupply: '21000000',
    };

    history.push('/coindetail', { coin });
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router history={history}>
            <Routes>
              <Route path="/coindetail" element={<CoinDetails />} />
            </Routes>
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  })
});