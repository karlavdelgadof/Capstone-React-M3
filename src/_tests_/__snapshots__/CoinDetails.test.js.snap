// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coins test snapshot for coins 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="coins-info"
      >
        <div
          class="coin-bg"
        />
        <div
          class="coin-text"
        >
          <div
            class="coin-name"
          >
            <h2>
              Bitcoin
            </h2>
            <span>
              (
              BTC
              )
            </span>
          </div>
          <div
            class="coin-m-stats"
          >
            <span
              class="price"
            >
              $
               
              USD
            </span>
            <span
              class="stats-t"
            >
              1
              <i>
                <img
                  alt=""
                  src="stats2.png"
                />
              </i>
            </span>
          </div>
        </div>
        <div
          class="stats-container"
        >
          <div
            class="stats-title"
          >
            Current Stats
          </div>
          <div
            class="stats-data"
          >
            <div
              class="stat-item"
            >
              <span>
                % Change (24h)
              </span>
              1.44
            </div>
            <div
              class="stat-item"
            >
              <span>
                % Change (1h)
              </span>
              -0.36
            </div>
            <div
              class="stat-item"
            >
              <span>
                % Change (7d)
              </span>
              -5.64
            </div>
            <div
              class="stat-item"
            >
              <span>
                BTC
              </span>
              1.00
            </div>
            <div
              class="stat-item"
            >
              <span>
                Market Cap $
              </span>
              385002452610.93
            </div>
            <div
              class="stat-item"
            >
              <span>
                Volume (24h)
              </span>
              27391301861.27469
            </div>
            <div
              class="stat-item"
            >
              <span>
                Volume (24h A)
              </span>
              28376681317.71335
            </div>
            <div
              class="stat-item"
            >
              <span>
                Coin Supply
              </span>
              19114346.00
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="coins-info"
    >
      <div
        class="coin-bg"
      />
      <div
        class="coin-text"
      >
        <div
          class="coin-name"
        >
          <h2>
            Bitcoin
          </h2>
          <span>
            (
            BTC
            )
          </span>
        </div>
        <div
          class="coin-m-stats"
        >
          <span
            class="price"
          >
            $
             
            USD
          </span>
          <span
            class="stats-t"
          >
            1
            <i>
              <img
                alt=""
                src="stats2.png"
              />
            </i>
          </span>
        </div>
      </div>
      <div
        class="stats-container"
      >
        <div
          class="stats-title"
        >
          Current Stats
        </div>
        <div
          class="stats-data"
        >
          <div
            class="stat-item"
          >
            <span>
              % Change (24h)
            </span>
            1.44
          </div>
          <div
            class="stat-item"
          >
            <span>
              % Change (1h)
            </span>
            -0.36
          </div>
          <div
            class="stat-item"
          >
            <span>
              % Change (7d)
            </span>
            -5.64
          </div>
          <div
            class="stat-item"
          >
            <span>
              BTC
            </span>
            1.00
          </div>
          <div
            class="stat-item"
          >
            <span>
              Market Cap $
            </span>
            385002452610.93
          </div>
          <div
            class="stat-item"
          >
            <span>
              Volume (24h)
            </span>
            27391301861.27469
          </div>
          <div
            class="stat-item"
          >
            <span>
              Volume (24h A)
            </span>
            28376681317.71335
          </div>
          <div
            class="stat-item"
          >
            <span>
              Coin Supply
            </span>
            19114346.00
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
